transport/split/stream_dialer.go

```diff
@@ -22,19 +22,21 @@ import (
 )
 
 type splitDialer struct {
-	dialer     transport.StreamDialer
-	splitPoint int64
+	dialer        transport.StreamDialer
+	splitPoint    int64
+	repeatsNumber int64
+	skipBytes     int64
 }
 
 var _ transport.StreamDialer = (*splitDialer)(nil)
 
 // NewStreamDialer creates a [transport.StreamDialer] that splits the outgoing stream after writing "prefixBytes" bytes
-// using [SplitWriter].
-func NewStreamDialer(dialer transport.StreamDialer, prefixBytes int64) (transport.StreamDialer, error) {
+// using [SplitWriter]. If "repeatsNumber" is not 0, will split that many times, skipping "skipBytes" in between packets.
+func NewStreamDialer(dialer transport.StreamDialer, prefixBytes int64, repeatsNumber int64, skipBytes int64) (transport.StreamDialer, error) {
```

Comment:
Let's make `splitDialer` public (`SplitDialer`) and return that in this function, and add setters for the repeat and skip. Keep the member variables private so people don't mess with them.

As a design philosophy, we would rather keep the constructor with only the required parameters, with optional options as setters. That keeps the code extensible and simple for those that don't care about the advanced options.

This approach will preserve backwards compatibility as well.

Reply:
Aargh, the setters don't work as proposed beecause we return different implementations depending on the parameters. Need to rethink...